### PR TITLE
FEAT: spice model terminal pair

### PR DESCRIPTION
**Pull request by ring630**

### CODE REVIEW

File: src/pyedb/dotnet/edb_core/cell/hierarchy/component.py [hunk 1]
```diff
@@ -808,7 +808,9 @@ def _set_model(self, model):  # pragma: no cover
             return False
         return True
 
-    def assign_spice_model(self, file_path, name=None, sub_circuit_name=None):
+    def assign_spice_model(
+        self, file_path: str, name: str = None, sub_circuit_name: str = None, terminal_pairs: list = None
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/cell/hierarchy/component.py:812]: If you are adding type hints, you should add optional inputs as `Optional[str]` and `Optional[list]` and adding associated import `from typing import Optional`

File: src/pyedb/dotnet/edb_core/cell/hierarchy/component.py [hunk 2]
```diff
@@ -828,23 +830,30 @@ def assign_spice_model(self, file_path, name=None, sub_circuit_name=None):
...
-            return False
+
+        model = self._edb.cell.hierarchy._hierarchy.SPICEModel()
+        model.SetModelPath(file_path)
+        model.SetModelName(name)
+        if sub_circuit_name:
+            model.SetSubCkt(sub_circuit_name)
+
+        if not len(pin_names_sp) == self.numpins:  # pragma: no cover
+            raise ValueError(f"Pin counts doesn't match component {self.name}.")
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/cell/hierarchy/component.py:845]: This should be moved before defining model, right ?

**Review by SMoraisAnsys — DISMISSED** [review]
LGTM, I left minor comments

File: src/pyedb/dotnet/edb_core/cell/hierarchy/component.py (hunk 2 above)
Comment (ring630) [reply src/pyedb/dotnet/edb_core/cell/hierarchy/component.py]: Good point

**Review by svandenb-dev — APPROVED** [review]
LGTM

**Comment by aohta2021** [comment]
I have confirmed SPICE model can assigned by Json config file.
Thank you!

**Pull request merged**